src/main/java/io/temporal/client/ActivityCompletionClient.java

```diff
@@ -19,41 +19,37 @@
 
 package io.temporal.client;
 
-import io.temporal.activity.Activity;
-import io.temporal.common.v1.WorkflowExecution;
-import io.temporal.failure.CanceledFailure;
+import io.temporal.activity.ActivityExecutionContext;
+import java.util.Optional;
 
 /**
- * Used to complete asynchronously activities that called {@link Activity#doNotCompleteOnReturn()}.
+ * Used to complete asynchronously activities that called {@link
+ * ActivityExecutionContext#doNotCompleteOnReturn()}.
  *
- * <p>Use {@link WorkflowClient#newActivityCompletionClient()} to create an instance. TODO: Throw
- * relevant exceptions like EntityNotExists, etc.
+ * <p>Use {@link WorkflowClient#newActivityCompletionClient()} to create an instance.
  */
 public interface ActivityCompletionClient {
 
   <R> void complete(byte[] taskToken, R result) throws ActivityCompletionException;
 
-  <R> void complete(WorkflowExecution execution, String activityId, R result)
+  <R> void complete(String workflowId, Optional<String> runId, String activityId, R result)
```

Comment:
It feels weird to have optional parameter in the middle of signature.  It is considered very bad practice to have required parameters after optional ones.